src/Compilers/CSharp/Test/Emit3/Semantics/CollectionExpressionArgumentTests.cs

```diff
@@ -0,0 +1,1408 @@
+                        y = [t, with(t)];
+                    }
+                }
+                """;
+            var comp = CreateCompilation([sourceA, sourceB]);
+            comp.VerifyEmitDiagnostics(
+                // (7,13): error CS0417: 'U': cannot provide arguments when creating an instance of a variable type
+                //         x = [with(t), t];
+                Diagnostic(ErrorCode.ERR_NewTyvarWithArgs, "[with(t), t]").WithArguments("U").WithLocation(7, 13),
+                // (8,13): error CS0417: 'U': cannot provide arguments when creating an instance of a variable type
+                //         x = [t, with(t)];
+                Diagnostic(ErrorCode.ERR_NewTyvarWithArgs, "[t, with(t)]").WithArguments("U").WithLocation(8, 13),
+                // (8,17): error CS9275: Collection argument element must be the first element.
+                //         x = [t, with(t)];
+                Diagnostic(ErrorCode.ERR_CollectionArgumentsMustBeFirst, "with").WithLocation(8, 17),
+                // (14,13): error CS0417: 'U': cannot provide arguments when creating an instance of a variable type
+                //         y = [with(t), t];
+                Diagnostic(ErrorCode.ERR_NewTyvarWithArgs, "[with(t), t]").WithArguments("U").WithLocation(14, 13),
+                // (15,13): error CS0417: 'U': cannot provide arguments when creating an instance of a variable type
+                //         y = [t, with(t)];
+                Diagnostic(ErrorCode.ERR_NewTyvarWithArgs, "[t, with(t)]").WithArguments("U").WithLocation(15, 13),
+                // (15,17): error CS9275: Collection argument element must be the first element.
+                //         y = [t, with(t)];
+                Diagnostic(ErrorCode.ERR_CollectionArgumentsMustBeFirst, "with").WithLocation(15, 17));
+        }
+
+        [Fact]
+        public void UnrecognizedType()
+        {
+            string source = """
+                class Program
+                {
+                    static A EmptyArgs() => [with()];
+                    static B NonEmptyArgs() => [with(default)];
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (3,12): error CS0246: The type or namespace name 'A' could not be found (are you missing a using directive or an assembly reference?)
+                //     static A EmptyArgs() => [with()];
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "A").WithArguments("A").WithLocation(3, 12),
+                // (4,12): error CS0246: The type or namespace name 'B' could not be found (are you missing a using directive or an assembly reference?)
+                //     static B NonEmptyArgs() => [with(default)];
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "B").WithArguments("B").WithLocation(4, 12));
+        }
+
+        [Fact]
+        public void EvaluationOrder_01()
```

Comment:
Could we include user-define conversions in this test, to observe the full order?